test/spec/analysis/analysis-model.spec.js

```diff
@@ -101,15 +118,101 @@ describe('src/analysis/analysis-model.js', function () {
 
       it('should reload the map', function () {
         this.analysisModel.set('type', 'something');
-        expect(this.vis.reload).toHaveBeenCalled();
+        expect(vis.reload).toHaveBeenCalled();
       });
 
       it('should keep listening type change again', function () {
         this.analysisModel.set('type', 'something');
-        expect(this.vis.reload).toHaveBeenCalled();
-        this.vis.reload.calls.reset();
+        expect(vis.reload).toHaveBeenCalled();
+        vis.reload.calls.reset();
         this.analysisModel.set('type', 'something else');
-        expect(this.vis.reload).toHaveBeenCalled();
+        expect(vis.reload).toHaveBeenCalled();
+      });
+    });
+
+    describe('on status change', function () {
+      var createAnalysisModelNoStatusNoReferences = function (visModel) {
+        var analysisModel = createFakeAnalysis({ id: 'a0' }, visModel);
+        return analysisModel;
+      };
+
+      var createAnalysisModelNoStatusWithReferences = function (visModel) {
+        var analysisModel = createFakeAnalysis({ id: 'a0' }, visModel);
+        analysisModel.markAsSourceOf(new Backbone.Model());
+        return analysisModel;
+      };
+
+      var createAnalysisModelWithStatusNoReferences = function (visModel) {
+        var analysisModel = createFakeAnalysis({ id: 'a0', status: 'foo' }, visModel);
+        return analysisModel;
+      };
+
+      var createAnalysisModelWithStatusWithReferences = function (visModel) {
+        var analysisModel = createFakeAnalysis({ id: 'a0', status: 'foo' }, visModel);
+        analysisModel.markAsSourceOf(new Backbone.Model());
+        return analysisModel;
+      };
+
+      var testCases = [
```

Comment:
this hacks usually give problems

Reply:
There shouldn't be any problem with these tests that don't have any `this` involved IMO. The question is... are they easy to understand?